Runtime/UIElements/Controls/UIUtils.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+static class UIUtils
+    {
+        static readonly string[] s_SizeUnits = { "KB", "MB", "GB", "TB" };
+
+        public static void SetElementDisplay(VisualElement element, bool value)
+        {
+            if (element == null)
+                return;
+
+            element.style.display = value ? DisplayStyle.Flex : DisplayStyle.None;
+            element.style.visibility = value ? Visibility.Visible : Visibility.Hidden;
+        }
+
+        public static void SetElementDisplayNonEmpty(TextElement element)
+        {
+            if (element == null)
+                return;
+
+            var nonEmpty = !String.IsNullOrEmpty(element.text);
+            element.style.display = nonEmpty ? DisplayStyle.Flex : DisplayStyle.None;
+            element.style.visibility = nonEmpty ? Visibility.Visible : Visibility.Hidden;
+        }
+
+        public static bool IsElementVisible(VisualElement element)
+        {
+            return element.resolvedStyle.visibility == Visibility.Visible && element.resolvedStyle.display != DisplayStyle.None;
+        }
+
+        public static void ScrollIfNeeded(ScrollView container, VisualElement target)
+        {
+            if (target == null || container == null)
+                return;
+
+            var containerWorldBound = container.worldBound;
+            var targetWorldBound = target.worldBound;
+
+            var minY = containerWorldBound.yMin;
+            var maxY = containerWorldBound.yMax;
+            var itemMinY = targetWorldBound.yMin;
+            var itemMaxY = targetWorldBound.yMax;
+
+            var scroll = container.scrollOffset;
+
+            if (itemMinY < minY)
+            {
+                scroll.y -= Math.Max(0, minY - itemMinY);
+                container.scrollOffset = scroll;
+            }
+            else if (itemMaxY > maxY)
+            {
+                scroll.y += itemMaxY - maxY;
+                container.scrollOffset = scroll;
+            }
+        }
+
+        public static IEnumerable<T> GetParentsOfType<T>(VisualElement element) where T : VisualElement
+        {
+            var result = new List<T>();
+
+            var parent = element;
+            while (parent != null)
+            {
+                if (parent is T selected)
+                    result.Add(selected);
+
+                parent = parent.parent;
+            }
+
+            return result;
+        }
+
+        public static T GetParentOfType<T>(VisualElement element) where T : VisualElement
```

Comment:
same as above.

Reply:
Also maybe rename to GetFirstParentOfType
And another idea is to actually add this extension to `UQueryBuilder` -> if that is possible this would be the preferred way to go.

P.S. Very needed utility method!